src/main/scala/io/digitallibrary/bookapi/service/translation/TranslationService.scala

```diff
@@ -31,21 +31,26 @@ trait TranslationService {
           readService.withIdAndLanguage(translateRequest.bookId, fromLanguage) match {
             case None => Failure(new NotFoundException())
             case Some(originalBook) => {
-              crowdinClientBuilder.forSourceLanguage(fromLanguage).flatMap(crowdinClient => {
-                val existingTranslations = translationDbService.translationsForOriginalId(translateRequest.bookId)
-
-                val toAddUser = existingTranslations.find(tr => tr.fromLanguage == fromLanguage && tr.toLanguage == LanguageTag(toLanguage))
-                val toAddLanguage = existingTranslations.find(tr => tr.fromLanguage == fromLanguage)
-
-                val inTranslationTry = (toAddUser, toAddLanguage) match {
-                  case (Some(addUser), _) => addUserToTranslation(addUser, translateRequest.userId.get)
-                  case (None, Some(addLanguage)) => addTargetLanguageForTranslation(addLanguage, translateRequest, originalBook.id, LanguageTag(originalBook.language.code), crowdinClient)
-                  case _ => createTranslation(translateRequest, originalBook, fromLanguage, toLanguage, crowdinClient)
+              readService.withIdAndLanguage(translateRequest.bookId, LanguageTag(toLanguage)).map(_.officiallyApproved).getOrElse(Some(false)) match {
```

Comment:
Kunne denne sjekken vore slått sammen med den forrige for å slippe to reads fra basen?

Reply:
Tenker du da ved å lage en ny spørring(withIdAndLanguageAndToLanguage) som returnerer to bøker?

Reply:
Eg såg berre at det var to reads, men dei gjør vel to forskjellige ting då.

Reply:
Ja, den ene sjekker at språket som oversettes fra er riktig, of den andre sjekker om du kan oversette til det nye språket.